src/jit/importer.cpp

```diff
@@ -16440,6 +16439,33 @@ bool Compiler::impReturnInstruction(BasicBlock* block, int prefixFlags, OPCODE&
 #endif
                 }
 
+                // If we are inlining a method that returns a struct byref, check whether we are "reinterpreting" the
+                // struct.
+                if ((returnType == TYP_BYREF) && (info.compRetType == TYP_BYREF) && (op2->OperGet() == GT_ADDR))
```

Comment:
It may be possible that `op2` is a `GT_COMMA `with `gtEffectiveVal()` that is a `GT_ADDR(GT_LCL_VAR)`

So using: 
```
GenTree* retVal = op2->gtEffectiveVal();
if ( .. (retVal->OperGet() == GT_ADDR))
```

would be better.

Reply:
Done.